### PR TITLE
TT-237 - Modal Component Unit Tests

### PR DESCRIPTION
**Pull request by MontgomeryWatts**
This ticket adds some unit tests to the frontend. I figured it would be a good place to start because it is nearly a functional component and has few edge cases.

### CODE REVIEW

File: test/unit/specs/components/Modal.spec.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  it('does not display empty message', () => {
+    const wrapper = shallowMount(Modal, {
+      propsData: {
+        responseMessage: ''
+      }
+    })
+    expect(wrapper.find('message').exists()).to.be.false
+  })
+
+  it('emits close event when background is clicked', async () => {
```
Comment (MontgomeryWatts) [inline test/unit/specs/components/Modal.spec.js]: Async needed to be specified because we have to wait for the Vue DOM to update after the click event.

File: test/unit/specs/components/Modal.spec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    const wrapper = shallowMount(Modal, {
+      propsData: {
+        showModal: true
+      }
+    })
+    wrapper.find('.modal-background').trigger('click')
+    await wrapper.vm.$nextTick()
```
Comment (MontgomeryWatts) [inline test/unit/specs/components/Modal.spec.js]: The update is awaited here.
